src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3272,6 +3270,16 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                             {
                                 result = ((long)Convert.ChangeType(d, typeof(long), CultureInfo.InvariantCulture)) * multiplier;
                             }
+                            if (suffix == 'u' || suffix == 'U') {
```

Comment:
>suffix == 'u' || suffix == 'U' [](start = 32, length = 30)

If we are adding `unsigned` support, it seems like we should also support `ul`.

Reply:
I thought it better to simply have the parser infer `ulong` from overlarge `<int>u` literals. But would it be better to have the `ul` explicitly permitted?

Are two-letter suffixes even supported currently?

Reply:
So in other words, what you're asking isn't really plausible without significantly changing how suffix detection works.

I would argue that is a more breaking change.

Reply:
It's a slightly more involved change, sure, but it's still a simple change and one that people will expect as long as you're adding unsigned support.

As for a breaking change, `u` is a breaking change as well and it's better to cover all expected use cases at once if taking the breaking change.

Reply:
Touche!

Okay, let me see if I can trace that down... `suffix` is referenced as a `char` value basically everywhere... This is gonna take some thought.

Reply:
c# uses one suffix 'u' for int and long
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/uint
> Integer literals can also include a suffix that denotes the type. The suffix U or 'u' denotes either a uint or a ulong, depending on the numeric value of the literal. The following example uses the u suffix to denote an unsigned integer of both types. Note that the first literal is a uint because its value is less than UInt32.MaxValue, while the second is a ulong because its value is greater than UInt32.MaxValue.

Reply:
@lzybkr consider this one done and dusted.

References replaced, quotation marks swapped out, null terminator accounted for (not even sure why that's necessary, but... it is! apparently! I am unsure why!), and it correctly parses `ul` without allowing strange nonsense suffixes like `ll` or anything even longer that it could theoretically now support.

Reply:
@iSazonov Good to know! We can always revert the two-character implementation of `suffix` if you feel two characters is too many. It *does* open up some possibilities, however. I'll leave that decisions up to you folks; I personally am perfectly happy with a single `u` although I do appreciate the parity of `l` and `ul`.

As for parsing *everything* as `BigInteger`... I don't think that's particularly efficient. Especially since half our numerics are `double` or `decimal`!

Additionally, parsing everything as `BigInteger` unnecessarily would a) go against the decision made by you and others in #7510 and b) likely increase the memory footprint of the parsing method appreciably (although admittedly I can only speculate on that).

Reply:
We use BigInteger here only for excluding an overflow not for expose it public.

As for memory footprint I think we should use ReadOnlySpan<char> - replace `string strNum` with `ReadOnlySpan<char> strNum` - all TryParse-s use ReadOnlySpan<T> internally.

Reply:
You can look https://source.dot.net/#System.Private.CoreLib/shared/System/Int64.cs,517dc8207c8fc99e
- in .Net Core all parse methods with string parameter is a wrapper for methods with ReadOnlySpan.

ReadOnlySpan is very useful for parsers. Here we can surely get benefits.

Reply:
I'm sure we can, but I'm afraid the changes required for *that* particular wrinkle might be above my abilities at present. I'll revisit the possibility after I take care of the other concerns raised and see what I can do

Reply:
@iSazonov from what I can see the parsing methodology is already by-each-character as it would be in that construction.

It's also building the string back up with `StringBuilder` before actually returning it, so it's going to be fairly efficient. 

I think I can say with reasonable confidence that at this point this may be superfluous. The only place that this could reasonably be done would be with the entire `_script` variable here. It may well offer some performant benefit if implemented here instead of making this a string. From what I can see there may be an implicit cast between `string` and `ReadOnlySpan<char>` which may allow me to simply type case the private variable and call it a day.

Everything I'm seeing interact with the _script data itself is actually pulling the data by index anyway, so... this may be a fairly simple thing too. But in terms of using it in *this* parsing segment it's not possible, because it's a read only span and cannot be built in the same way StringBuilder does as far as I know; this is necessary in order to pass the complete token around. 

I don't know that it makes sense to pass the token around as ReadOnlySpan<char> since it's already being built up as string (and I don't think there's an easy way to do such a thing with ReadOnlySpan<char>).

Your thoughts on this one, @lzybkr ?

Reply:
My suggestion is:
```c#
private static bool TryGetNumberValue(ReadOnlySpan<char> strNum, bool hex, bool real, char suffix, long multiplier, out object result)

....
strNum = strNum.Slice(1);
```

Reply:
There is no slicing happening in this method. There is no purpose to that. This would confer little to no benefit; `strNum` here in `TryGetNumberValue` is only passed directly to `TryParse` methods in sequence to try to get a value, and this will rarely be a long string.

The only way I can see this being of use to be more efficient is if we're already passing around slices from a `ReadOnlySpan<char>` that comes all the way from `_script`. If *that* becomes a `Span`, all the strings in this class that get passed around can derive from slices of it, which would work wonderfully, I'm sure.

But short of that I don't see a lot of reasons to change the `TryGetNumberValue` method in this manner. (If you want me to have a crack at converting everything over to the `ReadOnlySpan<char>` type from the `_script` basis onwards I think I can do that.)

Reply:
Although, since all the TryParse methods have a public overload in .NET Core 2.1 that accepts this data type it would save all of *that* extra data being cloned in memory.... okay. 

Now I see why you were saying that; I misunderstood your intention there. I still think it would be *far* more worthwhile to have `_script` begin as a `ReadOnlySpan` simply for the slicing that already happens on that level to be more efficient, but there may be other limitations with this type that I'm not familiar with as well, so... not sure.

I'll go ahead and get the ReadOnlySpan applied to the TryGetNumberValue...